app/editor/src/components/wysiwyg/Wysiwyg.tsx

```diff
@@ -110,10 +114,32 @@ export const Wysiwyg = <T extends object>({
     }
   };
 
+  const tagMatch = /\[.*?\]/g;
```

Comment:
You're doing a global search here.  We're not interested in square brackets through the text, only the end of the string.  You will also need to remove whitespace at the end.